### DISCUSSION TITLE
prevent state refresh at bottom.

### DISCUSSION THREAD

**Discussion by wibed** [petyosi/react-virtuoso#469] [Q&A] [answered]
i have expensive items rendered. which get rerendered once the scrollbar has reached the bottom.

is there a configuration to prevent rerendering the whole list as soon as the scrollbar has reached the bottom?

i tried to modify atBottomStateChange, rangeChanged, overscan, endReached properties but had no luck

**Comment by wibed** [comment, marked as answer]
there is a state change once the scrollbar hits the bottom. causing a rerender.
can one disable this?

**Reply by petyosi** [reply]
Can you show what you observe in a sandbox?

**Reply by wibed** [reply]
https://codesandbox.io/s/e6vmq?file=/src/index.js

if you click on one of the numbers the item appears. once scrolled down. all the items appeared disappear.
the item stay if you dont scroll all the way down, or cant scroll down at all

EDIT:
its weird in the codesandbox it disappears right away once it gets out of the view. which does not happen in the browser

**Reply by petyosi** [reply]
Are you sure this is the right sandbox? This is an example I did a while ago, no modifications on it

**Reply by wibed** [reply]
https://codesandbox.io/s/react-virutoso-with-react-beautiful-dnd-forked-7kn84?file=/src/index.js

woops here we go

**Reply by petyosi** [reply]
What I see in the sandbox is perfectly normal. Items are re-rendered as they move out of the viewport, so the DOM modification you do is lost.

**Reply by wibed** [reply]
in the ~~browser~~ , ran locally, the dom modification is preserved until the scrollbar hits the bottom.


furthermore, do you have a suggestion on how to preserve changes without using state or causing a rerender otherwise?

**Reply by wibed** [reply]
thanks for answering my question! 
ill work around it by pushing my elements in a useRef array.
